Gallery-Server/src/main/java/edu/yu/cs/gallery/ArtResource.java

```diff
@@ -27,69 +32,137 @@ public class ArtResource {
     @Inject 
     Utility utility;
 
-
+// ---------------------------------------Single-Requests--------------------------------------------//    
     @GET
     @Path("/{gallery-id}/arts")
     public Response getAll(
             @PathParam("gallery-id") long galleryId,
             @QueryParam("name") String name,
             @QueryParam("creator") String creator, @Context UriInfo uriInfo) throws URISyntaxException {
         if (utility.gallery == null || galleryId != utility.gallery.id) {
-            return utility.redirect(galleryId, uriInfo);
+            return utility.temporaryRedirect(galleryId, uriInfo);
         }
         return Response.status(Status.OK).entity(ar.search(galleryId, name, creator)).build();
     }
     
-    @GET
```

Comment:
was this replaced with something?

Reply:
No, we just organized the methods a bit differently to collect all the batch endpoints at the bottom of the class. See there.